openedx-unsupported/frontend-app-support-tools
# feat: allow runtime configuration

## Description

This PR is part of the work to make it possible to configure the frontend applications at runtime.

### Related
https://github.com/openedx/frontend-wg/issues/103
https://discuss.openedx.org/t/mfe-runtime-configuration/8495/1

## Changes

Update frontend-platform to version 2.5.0
Create a component Head that uses the Helmet library and integrates internationalization to change the MFE name in the title tag according to the language, and change the favicon in runtime.

## How to test

To allow runtime configuration, set new environment variables MFE_CONFIG_API_URL and APP_ID in the env file and add the API URL ( To test this, you can use the API from this https://github.com/openedx/edx-platform/pull/30473 or use an external tool to mock the API response).
The API should respond with a JSON with the config values, something like:
```
{
  "SITE_NAME": "Test Site",
  "LOGO_URL": "https://testimage.com/logo.svg",
  "LOGO_TRADEMARK_URL": "https://testimage.com/logo.svg",
  "LOGO_WHITE_URL": "https://testimage.com/logo.svg",
  "FAVICON_URL": "https://testimage.com/favicon.ico",
}
```
The initialize process should work normally.
Note: You can combine build time and runtime configuration

## Screenshot
![support tools evidence](https://user-images.githubusercontent.com/35668326/196480145-e0e6e598-d0ba-4d37-a517-38a4be1a88e3.png)
